variant_remapping_tools/reverse_strand.py

```diff
@@ -72,43 +20,114 @@
             cig_list.append(cig_type)
         begin += cig_length
     # Deletions need to be counted in the new variant position:
-    read_var_pos = start + flank_length + 1 + cig_list[0:flank_length+1].count(pysam.CDEL)
+    read_var_pos = start + flank_length + 1 + cig_list[0:flank_length + 1].count(pysam.CDEL)
     for operator in range(0, flank_length + 2 + local_region_size + cig_list[0:flank_length+1].count(pysam.CDEL)):
         # Stop incrementing varpos once we've reached read_var_pos:
         if start + operator < read_var_pos:
             # Mismatch/matches and deletions increase the position counter:
-            if cig_list[operator] == pysam.CMATCH or cig_list[operator] == pysam.CDEL: 
+            if cig_list[operator] == pysam.CMATCH or cig_list[operator] == pysam.CDEL:
                 varpos += 1
         # If we are in the local region around the variant, but not at the position of the variant:
-        if ((start + operator >= (read_var_pos - local_region_size)) 
-            and (start + operator <= (read_var_pos + local_region_size)) 
-            and (start + operator != (read_var_pos))):
+        if ((start + operator >= (read_var_pos - local_region_size))
+                and (start + operator <= (read_var_pos + local_region_size))
+                and (start + operator != read_var_pos)):
             if cig_list[operator] == 0:  # Match or mismatch
                 perf_counter += 1  # Increase the counter for perfect local region
-    if (AS <= int(score_cutoff)):
-        primary_poor_count += 1
-    elif (AS - XS < diff_cutoff):
-        gap_small_count += 1
-    elif (perf_counter != 2*local_region_size):
-        context_bad_count += 1
+    return varpos, perf_counter, local_region_size
+
+
+def is_read_valid(read, counter, flank_length, score_cutoff, diff_cutoff):
+    if read.is_secondary:  # We only want to deal with primary reads
+        return False
+    counter['total'] += 1
+    if read.is_unmapped:  # Can be decoded with bitwise flag with & 4 (4 means unmapped)
+        counter['unmapped'] += 1
+        return False
+    # Getting AS and XS:
+    AS = read.get_tag("AS")
+    try:
+        XS = read.get_tag("XS")  # Some variants don't have secondary alignments, which throws an error
```

Comment:
This is a super fine level nitpick, but currently there's a mixture of single and double quotes for string literals—maybe now's the time to fix it, before the module grows and it becomes more cumbersome

Reply:
(I think single quotes are better but that's a personal choice. PIP 8 just recommends the choice is consistent throughout the codebase.)

Reply:
That's true and not difficult to change. This is now done.